agents/neural_agent.py

```diff
@@ -68,7 +68,11 @@ def create_balanced_eval_set(cache: phyre.SimulationCache, task_ids: TaskIds,
     all_indices = np.concatenate([positive_indices, negative_indices])
     selected_actions = torch.FloatTensor(actions[all_indices % len(actions)])
     is_solved = torch.LongTensor(flat_statuses[all_indices].astype('int')) > 0
-    task_indices = torch.LongTensor(all_indices // len(actions))
+    
+    all_task_indices = np.arange(len(task_ids)).repeat(actions.shape[0])
+    positive_task_indices = all_task_indices[positive_indices]
+    negative_task_indices = all_task_indices[negative_indices]
+    task_indices = torch.LongTensor(np.concatenate([positive_task_indices, negative_task_indices]))
```

Comment:
Can we use `all_indices`?

```suggestion
    task_indices = torch.LongTensor(all_task_indices[all_indices])
```

Reply:
No, because `all_indices` is indices in the dataset (from 0 - 117,000,000) and `task_indices` is task indices compatible with phyre simulator (i.e. from 0 - 1600).
i guess all indices is an unused variable. Will remove it.